certificate-service/app/src/main/java/se/inera/intyg/certificateservice/application/certificate/service/CertificateDataConfigConverter.java

```diff
@@ -11,19 +14,20 @@
 public class CertificateDataConfigConverter {
 
   public CertificateDataConfig convert(ElementSpecification elementSpecification) {
-    return switch (elementSpecification.configuration().type()) {
-      case CATEGORY -> CertificateDataConfigCategory.builder()
+    if (CATEGORY.equals(elementSpecification.configuration().type())) {
+      return CertificateDataConfigCategory.builder()
           .text(elementSpecification.configuration().name())
           .build();
-      case DATE -> {
-        final var configuration = (ElementConfigurationDate) elementSpecification.configuration();
-        yield CertificateDataConfigDate.builder()
-            .id(configuration.id())
-            .text(configuration.name())
-            .minDate(configuration.minDate())
-            .maxDate(configuration.maxDate())
-            .build();
-      }
-    };
+    }
+    if (DATE.equals(elementSpecification.configuration().type())) {
+      final var configuration = (ElementConfigurationDate) elementSpecification.configuration();
+      return CertificateDataConfigDate.builder()
+          .id(configuration.id())
+          .text(configuration.name())
+          .minDate(configuration.minDate())
+          .maxDate(configuration.maxDate())
+          .build();
+    }
+    return null;
```

Comment:
If it reaches this I think it is better that we throw an exception than return null. For example, IllegalStateException which a message saying what type of configuration that didn't match an alternative. If it comes to this we have a bug in the code before and it is good to fail fast here.

Reply:
Makes sense, done!